Products/GenericSetup/context.py

```diff
@@ -519,12 +519,12 @@ def getSnapshotFolder( self ):
     security.declarePrivate( '_createObjectByType' )
     def _createObjectByType( self, name, body, content_type ):
 
-        if isinstance( body, unicode ):
+        if six.PY2 and isinstance(body, unicode):
             encoding = self.getEncoding()
             if encoding is None:
                 body = body.encode()
```

Comment:
Relying on the system default encoding is pretty much guaranteed to break something, eventually.